components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1577,12 +1614,12 @@ public IdentityProvider getEnabledIdPByRealmId(String realmId, String tenantDoma
      * @param identityProvider
      * @throws IdentityProviderManagementException
      */
-    private void handleMetadta(IdentityProvider identityProvider, StringBuilder idpEntityId, StringBuilder metadata) throws IdentityProviderManagementException {
+    private String handleMetadta(IdentityProvider identityProvider) throws IdentityProviderManagementException {
```

Comment:
Fix the typo in the name. Method name does not convey the functionality.

Reply:
created a seperate issue [1] to track metadata improvements. This issue will be fixed in it.
[1] https://github.com/wso2/product-is/issues/6750